test/RSocketTests.cpp

```diff
@@ -53,6 +54,58 @@ std::shared_ptr<RSocketClient> makeClient(
   return makeClientAsync(eventBase, port).get();
 }
 
+namespace {
+struct DisconnectedRequestResponseHandler : public rsocket::RSocketResponder {
+  DisconnectedRequestResponseHandler() {}
+
+  yarpl::Reference<yarpl::single::Single<rsocket::Payload>>
+  handleRequestResponse(rsocket::Payload, rsocket::StreamId) override {
+    assert(false);
```

Comment:
Nit: Shouldn't this be named `DisconnectedResponder`?  It's doing `assert(false)` on all possible requests, not just req-response.

Nit: I'd replace `assert(false)` with `CHECK` tbh, I think it makes sense to fail loudly in all build modes for this test-only construct.

Reply:
That's a better name yeah; current name is a relic from when it was just for R/R tests.